app/server/tests/controllers/appsController.test.ts

```diff
@@ -3,13 +3,18 @@ import { WodinWebError } from "../../src/errors/wodinWebError";
 
 // Need to mock getSessionStore before importing the controller
 let sessionIdFromFriendlyId: string | null;
-const mockSessionStore = {
-    getSessionIdFromFriendlyId: jest.fn().mockImplementation(() => { return sessionIdFromFriendlyId; })
-};
-const mockGetSessionStore = jest.fn().mockReturnValue(mockSessionStore);
-jest.mock("../../src/db/sessionStore", () => { return { getSessionStore: mockGetSessionStore }; });
+const { mockSessionStore, mockGetSessionStore } = vi.hoisted(() => {
+    const mockSessionStore = {
+        getSessionIdFromFriendlyId: vi.fn().mockImplementation(() => { return sessionIdFromFriendlyId; })
+    };
+    const mockGetSessionStore = vi.fn().mockReturnValue(mockSessionStore);
+    return {
+        mockSessionStore,
+        mockGetSessionStore
+    }
+})
+vi.mock("../../src/db/sessionStore", () => { return { getSessionStore: mockGetSessionStore }; });
```

Comment:
just to explain this pattern in a couple of places in this PR and future ones:
`vi.mock` works almost the same as `jest.mock` however it cannot capture external variables except for those that have been hoisted through `vi.hoisted` so `vi.hoisted` expressions are always run first so are guaranteed to exist for the mock, so most of it is the same, just any variable (such as a mocked method that you want to test) that is used in the tests and the mock must be hoisted

Reply:
umm are you sure?? ive never had to hoist like this when using vitest.. if your mock starts with mock it should auto hoist

Reply:
yep i think so, here i have tried to just define them as variables with prefix mock however i get this error
![image](https://github.com/user-attachments/assets/642f818f-db37-4bc0-9d1b-1fa534e04b53)

its because vi.mock itself is hoisted to the top so i need to make sure those mocks are also hoisted so they exist when vi.mock runs